### PR TITLE
Optimize AIP 12

### PR DESCRIPTION
**Pull request by wjmelements**
#### Changes
* removeAll: Fix transfer amount
* removeAll: Loop through set in reverse to make removal SSTOREs cheaper
* cache supportedTokenList in more contexts to save on keccaks

Reviewers @dmosites

### CODE REVIEW

**Comment by wjmelements** [comment]
Trying to get other optimizations into openzeppelin (we'll see about that lol) but I think this is ready for review.

**Comment by wjmelements** [comment]
The openzeppelin optimizations:

* [EnumerableSet: Clear](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2607)
* [EnumerableSet: Remove Boundary Check in _at](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2606)
* [EnumerableSet: Skip SSTORE On Last Item](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2605)

If any of these aren't accepted you can build from a fork that cherry-picks them.

File: source/registry/contracts/Registry.sol [hunk 1]
```diff
@@ -65,13 +69,14 @@ contract Registry is Ownable {
     EnumerableSet.AddressSet storage supportedTokenList =
       supportedTokens[msg.sender];
     uint256 length = supportedTokenList.length();
-    for (uint256 i = 0; i < length; i++) {
-      address token = supportedTokenList.at(0);
+    uint256 originalLength = length;
+    while (length --> 0) {
```
Comment (dmosites) [inline source/registry/contracts/Registry.sol:73]: Prefer `length-- > 0` for clarity.

Comment (wjmelements) [reply source/registry/contracts/Registry.sol]: worth noting: this behavior does not work in solidity 0.8 because it does not appreciate the underflow.

Comment (wjmelements) [reply source/registry/contracts/Registry.sol]: of course that can be worked around with `unchecked`

**Comment by wjmelements** [comment]
outdated; see https://github.com/airswap/airswap-protocols/pull/607

**Pull request closed without merging**